agent/src/main.rs

```diff
@@ -28,6 +46,46 @@ async fn read_hostname() -> Result<String> {
         .to_owned())
 }
 
+async fn read_ssh_pub_host_key(String key_location) -> Result<String> {
+    let public_key = tokio::fs::read_to_string(key_location)
+        .await
+        .context("error reading pub key")?
+        .trim_end()
+        .split(" ")
+        .nth(1)
+        .to_owned()
+    let pub_key_decoded = base64::decode(public_key).unwrap()
+    let mut sha256_hasher = Shae256::new()
+    sha256_hasher.update(pub_key_decoded)
+    let hashed_result = sha256_hasher.finalize()
+    Ok(str::from_utf8(simple_hex(&hashed_result)).chars().filter(|c| !c.is_whitespace()).collect())
```

Comment:
Yeah this doesn't at all look like something we want. Maybe try https://docs.rs/hex/0.4.2/hex/fn.encode.html

Reply:
Generally that filter & chars & everything is way more expensive than we need it to be.